client/src/util/stateManager/controlsHelpers.ts

```diff
@@ -31,107 +36,80 @@ Remember that option values can be ANY js type, except undefined/null.
   );
+  const categoryValueIndices = new Map(categoryValues?.map((v, i) => [v, i]));
   const numCategoryValues = categoryValueIndices.size;
 
   return {
-    allCategoryValues, // array: of natively typed category values (all of them)
-    categoryValues, // array: of natively typed category values (top N only)
-    categoryValueIndices, // map: category value (native type) -> category index (top N only)
-    numCategoryValues, // number: of values in the category (top N)
-    categoryValueCounts, // array: cardinality of each category, (top N)
-    isUserAnno, // bool
+    allCategoryValues,
+    categoryValues,
+    categoryValueIndices,
+    numCategoryValues,
+    categoryValueCounts,
+    isUserAnno,
   };
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-export function createCategoricalSelection(names: any) {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  return fromEntries(names.map((name: any) => [name, new Map()]));
-}
-
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-export function pruneVarDataCache(varData: any, needed: any) {
-  /*
-  Remove any unneeded columns from the varData dataframe.  Will only
-  prune / remove if the total column count exceeds VarDataCacheLowWatermark
-
-  Note: this code leverages the fact that dataframe offsets indicate
-  the order in which the columns were added.  This crudely provides
-  LRU semantics, so we can delete "older" columns first.
-  */
-
-  /*
-  VarDataCacheLowWatermark - this config value sets the minimum cache size,
-  in columns, below which we don't throw away data.
-
-  The value should be high enough so we are caching the maximum which will
-  "typically" be used in the UI (currently: 10 for diffexp, and N for user-
-  specified genes), and low enough to account for memory use (any single
-  column size is 4 bytes * numObs, so a column can be multi-megabyte in common
-  use cases).
-  */
-  const VarDataCacheLowWatermark = 32;
-
-  const numOverWatermark = varData.dims[1] - VarDataCacheLowWatermark;
-  if (numOverWatermark <= 0) return varData;
-
-  const { colIndex } = varData;
-  const all = colIndex.labels();
-  const unused = difference(all, needed);
-  if (unused.length > 0) {
-    // sort by offset in the dataframe - ie, psuedo-LRU
-    unused.sort((a, b) => colIndex.getOffset(a) - colIndex.getOffset(b));
-    const numToDrop =
-      unused.length < numOverWatermark ? unused.length : numOverWatermark;
-    for (let i = 0; i < numToDrop; i += 1) {
-      varData = varData.dropCol(unused[i]);
-    }
-  }
-  return varData;
+export function createCategoricalSelection(names: string[]): {
+  [key: string]: Map<unknown, unknown>;
```

Comment:
Based off of [this comment describing CategoricalSelection](https://github.com/chanzuckerberg/single-cell-explorer/blob/8d21a5553ef33f37224b256fcf3f094efac8d8d7/client/src/reducers/categoricalSelection.ts#L3-L13), the Map can be typed so:
```suggestion
  [key: string]: Map<string, boolean>;
```

Reply:
Oh thank you for linking that! Updated 🙆‍♂️